salt/_states/metalk8s_cordon.py

```diff
@@ -70,4 +69,26 @@ def node_uncordoned(name, **kwargs):
     name
         the name of the node
     '''
+    # k-c-m 1.12 adds taint `node.kubernetes.io/unschedulable` when cordonning
+    # a node but k-c-m 1.11 does not remove this taint when uncordonning so,
+    # we need to manually remove it when necessary by patching taints.
+
+    all_node_taints = __salt__['metalk8s_kubernetes.node_taints'](
+        name, **kwargs
+    )
+    key_to_remove = "node.kubernetes.io/unschedulable"
+
+    if key_to_remove in (taint['key'] for taint in all_node_taints):
+        new_taints = [
+            taint for taint in all_node_taints
+            if taint['key'] != key_to_remove
+        ]
+
+    if new_taints != all_node_taints:
```

Comment:
You don't need this `if` as if you are in the previous `if` you need to "remove" the taint

Reply:
ok